File: c++ version/401-500/452-minimum-number-of-arrows-to-burst-balloons.cpp
int findMinArrowShots(vector<vector<int>>& points) {
    if (points.size() == 0) return 0;
    int num = 1;
    sort(points.begin(), points.end(),
        [](const vector<int> &b1, const vector<int> &b2) {
            return b1[1] < b2[1];
        });
    vector<int> prev = points[0], curr;
    for (int i = 1; i < points.size(); ++i) {
        curr = points[i];
        if (curr[0] > prev[1]) {
            ++num;
            prev = curr;
        }
    }
    return num;
}

File: python version/452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key = lambda x: x[1])
        count = 1
        prev = 0
        for curr in range(1, len(points)):
            if points[curr][0] > points[prev][1]:
                count += 1
                prev = curr
            
        return count